File: PSET1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Dollars: ");
    } while (dollars <= 0);

    int cents = round(dollars * 100);

    int coins = 0;

    while(cents >= 25)
    {
        coins += 1;
        cents -= 25;
    }
    while(cents < 25 && cents >= 10)
    {
        coins += 1;
        cents -=10;
    }
    while(cents < 10 && cents >= 5)
    {
        coins += 1;
        cents -= 5;
    }
    while(cents <5 && cents >0)
    {
        coins += 1;
        cents -= 1;
    }
    printf("%i\n", coins);
}

File: PSET6/cash.py
from cs50 import get_float

def main():
    while True:
        change = get_float("Change owed: ")
        if change >= 0:
            break
    change = change * 100
    coins = 0
    while change >= 25:
        coins += 1
        change -= 25
    while change < 25 and change >= 10:
        coins += 1
        change -= 10
    while change < 10 and change >= 5:
        coins += 1
        change -= 5
    while change <5 and change >0:
        coins += 1
        change -= 1
    print(coins)
main()


